lib/wallaroo/application.pony

```diff
@@ -198,6 +198,10 @@ class Pipeline[Out: Any val] is BasicPipeline
       Pipeline[Out](_stages, _dag_sink_ids)
     end
 
+  fun ref collect(): Pipeline[Out] =>
+    let collect_key = "collect-key-" + RoutingIdGenerator().string()
```

Comment:
Perhaps the magic string "collect-key-" could be taken out. Maybe the whole thing could be a primitve?
```
primitive CollectKeyIdGenerator
  apply() : String =>
    "collect-key-" + RoutingIdGenerator().string()
```

Reply:
Ok, I'll add something like this.